tests/epyccel/recognised_functions/test_numpy_types.py

```diff
@@ -0,0 +1,1234 @@
+    from numpy import int16, shape
+    a = int16(arr)
+    s = shape(a)
+    return len(s), s[0], s[1], a[0,0], a[1,0]
+
+@types('bool[:,:]')
+@types('int[:,:]')
+@types('int8[:,:]')
+@types('int16[:,:]')
+@types('int32[:,:]')
+@types('int64[:,:]')
+@types('float[:,:]')
+@types('float32[:,:]')
+@types('float64[:,:]')
+def get_int8_arr_2d(arr):
+    from numpy import int8, shape
+    a = int8(arr)
+    s = shape(a)
+    return len(s), s[0], s[1], a[0,0], a[1,0]
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Arrays not handled yet."),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = [
+            pytest.mark.python]
+        )
+    )
+)
+
+@pytest.mark.parametrize( 'get_int', [get_int64_arr_2d, get_int32_arr_2d, get_int16_arr_2d, get_int8_arr_2d])
+def test_numpy_int_array_like_2d(language, get_int):
+
+    size = (2, 5)
+
+    bl = randint(0, 1, size=size, dtype= bool)
+
+    integer8 = randint(min_int8, max_int8, size=size, dtype=np.int8)
+    integer16 = randint(min_int16, max_int16, size=size, dtype=np.int16)
+    integer = randint(min_int, max_int, size=size, dtype=np.int)
+    integer32 = randint(min_int32, max_int32, size=size, dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=size, dtype=np.int64)
+
+    fl = uniform(min_float / 2, max_float / 2, size = size)
+    fl32 = uniform(min_float32 / 2, max_float32 / 2, size = size)
+    fl32 = np.float32(fl32)
+    fl64 = uniform(min_float64 / 2, max_float64 / 2, size = size)
+
+    epyccel_func = epyccel(get_int, language=language)
+
+    assert epyccel_func(bl) == get_int(bl)
+    assert epyccel_func(integer8) == get_int(integer8)
+    assert epyccel_func(integer16) == get_int(integer16)
+    assert epyccel_func(integer) == get_int(integer)
+    assert epyccel_func(integer32) == get_int(integer32)
+    # the if block should be removed after resolving (https://github.com/pyccel/pyccel/issues/735).
+    if sys.platform != 'win32':
+        assert epyccel_func(integer64) == get_int(integer64)
+        assert epyccel_func(fl) == get_int(fl)
+        assert epyccel_func(fl64) == get_int(fl64)
+        # Python returns always -32768 wich is less than min_int8, epyccel function returns 0.
+        if get_int != get_int16_arr_2d: # pylint: disable=comparison-with-callable
+            assert epyccel_func(fl32) == get_int(fl32)
+
+@types('bool')
+@types('int')
+@types('int8')
+@types('int16')
+@types('int32')
+@types('int64')
+@types('float')
+@types('float32')
+@types('float64')
+def get_float(a):
+    from numpy import float as NumpyFloat
+    b = NumpyFloat(a)
+    return b
+
+@types('bool')
+@types('int')
+@types('int8')
+@types('int16')
+@types('int32')
+@types('int64')
+@types('float')
+@types('float32')
+@types('float64')
+def get_float64(a):
+    from numpy import float64
+    b = float64(a)
+    return b
```

Comment:
I really want to use a template for these. I don't remember if that means that the whole module would need to be translated at once though or if we could define the template once and still translate the functions individually. Do you know?

Reply:
i dont have much experience with templates to tell you.

Reply:
ok. Let's assume we can't for now then :(